python/mlcroissant/mlcroissant/_src/structure_graph/nodes/field.py

```diff
@@ -40,6 +40,20 @@ class Field(Node):
 
     JSONLD_TYPE = constants.ML_COMMONS_FIELD_TYPE
 
+    array_shape: list[int] | None = mlc_dataclasses.jsonld_field(
```

Comment:
nit: I would have `array_shape` mirror exactly `cr:arrayShape` (i.e., `str | None`). Then, I would have `array_shape_list` be a method of `Field` that has the logic.

```python
def array_shape_list(self) -> Sequence[int]:
  if self.is_array and not self.array_shape is None:  # example: this is not trivial logic
    return (-1,)
  # etc
```